Password Reset/Change #29

bryanchriswhite opened the pull request:
### Password reset flow:

User is at bridge-gui login page (click reset password link) —> user is taken to password reset form (type email and new password, and submit) —> user is taken to a confirmation page with a "check your email ..." message.

User gets the email, clicks the link which hits bridge api and then redirects back to bridge-gui login page. The login page shows a "success ... please log in." message (below).

<img width="515" alt="screen shot 2016-06-29 at 11 10 50 am" src="https://cloud.githubusercontent.com/assets/600733/16446887/3f6595b0-3dea-11e6-835c-49ac47442d9d.png">

bryanchriswhite commented:
This PR depends on [Bridge PR #127](https://github.com/Storj/bridge/pull/127)

MrFancyMonocle commented on src/containers/Auth/LoginForm/index.js:
```diff
@@ -18,42 +18,63 @@ import {reduxForm} from 'redux-form';
 export default class LoginForm extends Component {
+  static propTypes = {
+    fields: PropTypes.object.isRequired,
+    error: PropTypes.object,
```
Getting a warning.js:45Warning: Failed propType: Invalid prop `error` of type `string` supplied to `LoginForm`, expected `object`. Check the render method of `ReduxForm(LoginForm)` here.

MrFancyMonocle commented on src/routes.js:
```diff
@@ -2,32 +2,35 @@ import React from 'react';
+  FileBucket,
+  NotFound,
+  Support
+  // Billing,
```
Can remove these non-existent routes entirely; feel free to remove any stale references.

MrFancyMonocle commented on src/utils/apiClient.js:
```diff
@@ -55,17 +55,15 @@ class WrappedClient extends Client {
-    var iteration = 0;
-    var chunks = [];
-    var self = this;
+    const chunks    = [];
```
Let's go ahead and remove all aligned assignment spacing as we go through the project over time; it's proving awkward to maintain as code changes, even if it's pretty to view.

MrFancyMonocle commented:
Added a few notes inline...

bryanchriswhite commented:
@MrFancyMonocle nice catches. I've addressed those in 511f309.

I also noticed a hunk in `apiClient` that was accidentally omitted; I squashed it into cecc0d7. This omission was due to the need for several env tweaks to get bridge-gui to play nice with a local bridge server. I'll clean-up these changes and submit a second PR for them. I'll also open an issue for that in the mean time.
